client/components/LinkPreview.vue

```diff
@@ -234,8 +234,21 @@ export default {
 			});
 		},
 		updateShownState() {
-			// User has manually toggled the preview, do not apply default
-			if (this.link.shown !== null) {
+			// Ensure we don't have empty strings in the list of preview filters
+			const filterTokens = this.$store.state.settings.hidePreviewFilters
```

Comment:
can you make this its own function and write tests for edge cases?

Reply:
Absolutely I can do that